### PR TITLE
Feature/#56 bitbucket pipeline variant

### LINKED ISSUE #56: Feature: add pipelines config variant
We are using Bitbucket pipelines for CI regularly now. It would be helpful to have a well commented starter config

### PR DESCRIPTION
**Pull request by fractalcactus**
Fixes #56

### CODE REVIEW

**Comment by fractalcactus** [comment]
In the `apply_variant` method, I'm not sure where `ENV.fetch` is referring to? I see that I need to add `bitbucket_pipelines` to some enumerable constant
```
def apply_variant?(name)
  return true if ENV.fetch('VARIANTS', '').split(',').include?(name.to_s)

  ask_with_default("Add #{name} to this application?", :blue, 'N').downcase.start_with?('y')
end
```

File: variants/bitbucket-pipelines/ bitbucket-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            script:
+              - apk add --no-cache git
+              - git checkout -b deploy-uat
+              - git push -f https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP_NAME.git deploy-uat:master
+        - step:
+            deployment: Demo
+            script:
+              - apk add --no-cache git
+              - git checkout -b deploy-demo
+              - git push -f https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP_NAME.git deploy-demo:master
```
Comment (G-Rath) [inline variants/bitbucket-pipelines/ bitbucket-pipelines.yml:42]:
```suggestion
```

Demo envs are not a typical thing, and require a lot of bribing of CC 😉

**Comment by eoinkelly** [comment]
@fractalcactus Not sure if this answers your question but `ENV` is how Ruby makes environment variables (passed into the ruby process by it's parent process, usually a shell) available to ruby code. That line is checking for an environment variable called `VARIANTS` and then performing some stringy operations on it.

**Comment by eoinkelly** [comment]
Can somebody who has used bitbucket-pipelines recently comment on whether this yaml looks right (I can't do a meaningful review without doing a whole lot of research)

**Comment by G-Rath** [comment]
@eoinkelly frankly, I'm preparing to pull the trigger on finally moving us off bitbucket, so don't know if this is worth having now

**Pull request closed without merging**